### PR TITLE
Updated scrypto dependency to latest version

### PR DESCRIPTION
**Pull request by mstrug-rdx**
## Summary
Update Scrypto dependency to latest version required for:
- `Decimal` to/from bytes (PR https://github.com/radixdlt/radix-engine-toolkit/pull/118)
- `zeroize` on `PrivateKey`

Special attention to: `Locker` blueprint addition and `VersionedSchema` changes.

### CODE REVIEW

File: crates/radix-engine-toolkit/src/transaction_types/traverser/types/general.rs [hunk 1]
```diff
@@ -144,7 +144,8 @@ impl GeneralDetector {
                         | EntityType::GlobalAccessController
                         | EntityType::GlobalOneResourcePool
                         | EntityType::GlobalTwoResourcePool
-                        | EntityType::GlobalMultiResourcePool => FnRules::all_disallowed(),
+                        | EntityType::GlobalMultiResourcePool
+                        | EntityType::GlobalAccountLocker => FnRules::all_disallowed(),
```
Comment (0xOmarA) [inline crates/radix-engine-toolkit/src/transaction_types/traverser/types/general.rs:148]: Based on the discussion in Slack this should be in `FnRules::all_allowed()`. Can we please change that and also add a test?

File: crates/radix-engine-toolkit/tests/transaction_types.rs [hunk 2]
```diff
@@ -446,7 +450,8 @@ fn transfers_with_try_deposit_or_refund_are_invalid() {
 #[test]
 fn lock_fee_is_recognized_as_a_reserved_instruction1() {
     // Arrange
-    let mut test_runner = TestRunnerBuilder::new().without_trace().build();
+    let mut test_runner =
```
Comment (0xOmarA) [inline crates/radix-engine-toolkit/tests/transaction_types.rs:453]: Can we do a global find and replace for `test_runner` with `ledger` to be consistent with `radixdlt-scrypto`?

File: crates/radix-engine-toolkit/Cargo.toml [hunk 3]
```diff
@@ -8,14 +8,13 @@ build = "build.rs"
 # radixdlt-scrypto dependencies.
 sbor = { workspace = true }
 scrypto = { workspace = true }
-transaction = { workspace = true }
+radix-common = { workspace = true }
```
Comment (0xOmarA) [inline crates/radix-engine-toolkit/Cargo.toml:11]: Let's bump the version to `2.1.0-dev1` since this is not compatiable with either mainnet or stokenet

File: crates/radix-engine-toolkit-uniffi/src/builder/manifest_builder/builder.rs [hunk 4]
```diff
@@ -1056,24 +1055,24 @@ macro_rules! manifest_args {
     ($($args: expr),*$(,)?) => {{
         use ::sbor::Encoder;
         let mut buf = ::sbor::rust::vec::Vec::new();
-        let mut encoder = radix_engine_common::data::manifest::ManifestEncoder::new(
+        let mut encoder = radix_common::data::manifest::ManifestEncoder::new(
```
Comment (0xOmarA) [inline crates/radix-engine-toolkit-uniffi/src/builder/manifest_builder/builder.rs:1058]: Can you please add support for the account locker methods to the builder? Examples of how we've done this in the past is at the end of the file.

Please let me know if you think that this should be a separate PR to get this in quicker for Lukasz.

File: crates/radix-engine-toolkit/Cargo.toml (hunk 3 above)
Comment (mstrug-rdx) [reply crates/radix-engine-toolkit/Cargo.toml:11]: I'm getting `ld: malformed 64-bit a.b.c.d.e version number: 2.1.0-dev1` error on my MacOS.

Comment (mstrug-rdx) [reply crates/radix-engine-toolkit/Cargo.toml:11]: Ok, figured this out, it was due to custom `build.rs` for MacOS which adds library version through `ld` linker additional param. It is not mandatory so I can remove that part.

File: crates/radix-engine-toolkit-uniffi/src/builder/manifest_builder/builder.rs (hunk 4 above)
Comment (mstrug-rdx) [reply crates/radix-engine-toolkit-uniffi/src/builder/manifest_builder/builder.rs:1058]: I've added all of the `AccountLocker` blueprint functions, but there is an issue with `account_locker_airdrop` in `Swift` error: `Address` does not conform to protocol `Hashable`

Comment (0xOmarA) [reply crates/radix-engine-toolkit-uniffi/src/builder/manifest_builder/builder.rs:1058]: I've solved this issue by making the HashMap's key a string. Will need to look into getting a Hashable address soon.
